src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -338,7 +338,7 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
 
                 ImmutableHashSet<string> currentGlobsToIgnore = globsToIgnoreStack == null ? globsToIgnore : globsToIgnoreStack.Peek();
 
-                Dictionary<string, UpdateOperation> itemsWithNoWildcards = new Dictionary<string, UpdateOperation>(FileUtilities.GetIsFileSystemCaseSensitive() ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
+                Dictionary<string, UpdateOperation> itemsWithNoWildcards = new Dictionary<string, UpdateOperation>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This can possibly break linux here. If on disk, you have both `a` and `A`, this will break matching them properly there.

Reply:
I believe it should be case-insensitive on any platform. There's a bug about that [here](https://github.com/dotnet/msbuild/issues/5749).